storage/src/smt/utils.rs

```diff
@@ -82,7 +82,7 @@ macro_rules! get_sub_leaves {
         cf_iter
             .into_iter()
             .filter_map(|(k, v)| {
-                if key_len != k.len() {
+                if key_len != k.len() || k[..prefix_len] != $prefix[..] {
```

Comment:
Would it be better to use `take_while` so that the iteration stops after the first key that does not starts with the prefix (instead of having to filter all keys after the prefix)?

Reply:
please check again